build.gradle

```diff
@@ -297,12 +298,23 @@ subprojects {
         apply plugin: 'com.netflix.nebula.project'
 
         if (project.name != 'micrometer-bom') {
+            apply plugin: 'biz.aQute.bnd.builder'
+
             jar {
                 manifest.attributes.put('Automatic-Module-Name', project.name.replace('-', '.'))
                 metaInf {
                     from "$rootDir/LICENSE"
                     from "$rootDir/NOTICE"
                 }
+
+                bundle {
+                    // workaround for multi-version JARs
+                    //  see https://github.com/bndtools/bnd/issues/2227
+                    bnd '''\
+-fixupmessages: '^Classes found in the wrong directory: .*'
+-exportcontents: io.micrometer.*
+'''
+                }
```

Comment:
THe build is failing (should fail for you locally) since `micrometer-osgi-test` is a new module and gradle tries to download it for binary compatibility check.
Could you please exclude this, I can show you how:

In line 336 (I can't make a comment on that line), you can find this: https://github.com/micrometer-metrics/micrometer/blob/533bae08a95009b9dccf3afae5e139e25b0be4e4/build.gradle#L336

Could you please add the new project name to the empty list, so that it should be:
```groovy
if (!(project.name in ['micrometer-osgi-test'])) { 
```

Reply:
Thanks @jonatan-ivanov .

I updated the build and fixed up some checkstyle issues as well.
The build works locally and I can see the circle-ci jdk11 also passes, but the jdk17 one is failing. I will try to have a look at that shortly.

Reply:
@jonatan-ivanov, the Java 17 build is working as well now. The only failing one is the CLA check. I submitted the signed CLA a while ago, but not sure if I'm missing something in the process.

Reply:
Thanks for getting the build passing!

> I submitted the signed CLA a while ago, but not sure if I'm missing something in the process.

Did you sign the individual CLA or the corporate CLA? I checked your GitHub username under the individual CLA and a signature was not found.

Reply:
We signed the Corporate CLA (though https://cla.pivotal.io/sign/pivotal?repositoryId=micrometer-metrics/micrometer&pullRequestId=3457). I asked our legal team for confirmation, and got the following screenshot:

![image](https://user-images.githubusercontent.com/5557551/221909568-6afadd20-f7bc-47a8-909a-4887f1a58295.png)

Let me know if there is anything else I can check.

Reply:
Thanks for checking. For corporate CLAs it is either done via the company email domain or a GitHub organization. See [the FAQ](https://cla.pivotal.io/about) toward the bottom. If done via the email domain, an email address with the company email domain name needs to be associated with your GitHub account. If done via your company's GitHub organization, your GitHub account needs to be a member of that organization and that membership has to be public so the CLA tool can see you are part of that GitHub org.

Reply:
Hi @shakuzen .
I'm still struggling a bit to validate this.

The CLA was signed by GitHub user [rburnsr3](https://github.com/rburnsr3) on behalf of R3. 
We are both part of the [corda GitHub organisation](https://github.com/corda/).
Let me know if you can find a reference to that, or if there is another action on our side.

Thanks.

Reply:
It looks like your membership to the Corda organization is not public. Check, for example, [your GitHub profile](https://github.com/driessamyn) in a private browser. If you make your membership to the Corda org public so the CLA bot can see it, I think things will work.

Reply:
Thanks @shakuzen 
For some reason I thought that was the default setting.
I have updated my profile and the organisation should now be public.

Reply:
I'm glad we got it sorted out. Thank you for your patience and efforts.